### ISSUE TITLE
Error (171000): Can't fit design in device (DE0 Nano)

### ISSUE THREAD

**Issue by Andy2No** [jotego/jt12#69]
This probably ought to be a discussion, rather than an issue, but there isn't currently a Discussion tab on this repo.

I just tried to build jt12 for a DE0-Nano, using Quartus Prime Lite 21.1.  I added all .v files in the hdl folder and its subfolders to the project. The first problem I've encountered is that it claims the design uses more I/O pads than are available on the device:

Error (169282): There are 246 IO output pads in the design, but only 139 IO output pad locations available on the device.

Error (171000): Can't fit design in device

I was hoping to maybe move this to a smaller FPGA, once I got it working, but I'm surprised that it won't fit on the Cyclone IV DE0-Nano, which has an EP4CE22F17C6N with ~22,000 LEs.

So, what sort of hardware is needed or expected, to get this to build?  Is there any way to get the IO count closer to the 24 pin chip it emulates?

I bought what was claimed to be a Yamaha YM3438 a while ago, and set up a test circuit with an Atmega1284P microcontroller, earlier today.  I've concluded that the chip is fake (connecting what should be the clock pin puts so much load on the pin driving it that it can't power an LED through a 1k resistor at the same time - so it's probably the Vcc pin of a relabelled chip).  I'd looked at this repo before but wasn't at all clear how to build it.  I was hoping to be able to use an FPGA to emulate a YM3438 or YM2612 as an in circuit replacement.  Is that possible with this core?

I only have a very basic understanding of FPGAs and Verilog, but it's my long term aim to learn more.

**Comment by jotego** [comment]
I think you're doing something very wrong in your setup.
JT12 has been used for what you're trying to do before successfully. 
I trust you'll get it right with some work and dedication.

**Comment by Andy2No** [comment]
I just added all the files to a new project, and compiled it.  In the absence of any further instructions, I don't know what else I would have done.

Have other people mentioned how to build it?

**Comment by jotego** [comment]
It's used all over the place. Check it here. 

https://github.com/jotego/jtcores

This is not a Q&A for learning Intel Quartus. Please ask about specific problems related to the repository.

**Comment by Andy2No** [comment]
Specifically, I can build jt51, but not jt12.  I found 46 results by googling jotega jt12, and some of them were just you asking for help designing it, in forums - and as far as I can tell, not getting any.  I found nothing to help someone get started using it, or anyone discussing building it, only some praise from people using things pre-built with it.

No problem.  I'll look at other FPGA FM projects instead.
